Foundation/NSStream.swift

```diff
@@ -118,19 +139,34 @@ public class InputStream: Stream {
     
     // returns YES if the stream has bytes available or if it impossible to tell without actually doing the read.
     public var hasBytesAvailable: Bool {
-        NSUnimplemented()
+        return CFReadStreamHasBytesAvailable(_stream)
     }
     
     public init(data: Data) {
-        NSUnimplemented()
+        let nsData = data._bridgeToObjectiveC()
```

Comment:
Using NSData as an intermediate to convert the Data object to the expected CFData. Currently, this is the available option. Found a work item which says so: https://bugs.swift.org/browse/SR-1797